p2p/pkg/notifications/notifications.go

```diff
@@ -0,0 +1,66 @@
+package notifications
+
+import (
+	"github.com/cskr/pubsub/v2"
+)
+
+const (
+	TopicPeerConnected    = "peer_connected"
+	TopicPeerDisconnected = "peer_disconnected"
+)
+
+// Notification is a struct that represents a notification. It has a Topic field
+// that represents the topic of the notification and a Value field that represents
+// the value of the notification. The Value field is a map[string]any, which means
+// that it can be any type.
+type Notification struct {
+	Topic string
+	Value map[string]any
+}
+
+// Notifier is an interface that is used to notify about a notification. It will be
+// used by the publisher to notify the subscribers about the notification.
+type Notifier interface {
+	Notify(*Notification)
+}
+
+// Notifiee is an interface that is used to subscribe to a notification. It will be
+// used by the subscriber to subscribe to the notification.
+type Notifiee interface {
+	Subscribe(topics ...string) chan *Notification
```

Comment:
The topics param should be typed (not string) so only topics enumerated above can be passed.